From fb0a6d3141e1c9615fc4885a7cea975c2565afbc Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:05 -0400
Subject: [tools] perf/trace: Use socket's beautifiers in socketpair

Message-id: <1408462094-14194-137-git-send-email-jolsa@redhat.com>
Patchwork-id: 88127
O-Subject: [PATCH RHEL7.1 BZ1131394 136/325] perf trace: Use socket's beautifiers in socketpair
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 07120aa5d5a241895bcca8b6e5b5b8d7b1df1f2f
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Fri Sep 20 12:24:20 2013 -0300

description
===========
For the address family and socket type.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 4b30ec7..245534f 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -652,6 +652,10 @@ static struct syscall_fmt {
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* family */
         [1] = SCA_SK_TYPE, /* type */ },
    .arg_parm  = { [0] = &strarray__socket_families, /* family */ }, },
+ { .name     = "socketpair", .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_STRARRAY, /* family */
+        [1] = SCA_SK_TYPE, /* type */ },
+   .arg_parm  = { [0] = &strarray__socket_families, /* family */ }, },
  { .name     = "stat",     .errmsg = true, .alias = "newstat", },
  { .name     = "tgkill",     .errmsg = true,
    .arg_scnprintf = { [2] = SCA_SIGNUM, /* sig */ }, },
-- 
1.7.1